test/Confluent.Kafka.IntegrationTests/Tests/AddBroker.cs

```diff
@@ -0,0 +1,108 @@
+        /// </summary>
+        [Theory, MemberData(nameof(KafkaParameters))]
+        public static void AddBrokers(string bootstrapServers, string topic, string partitionedTopic)
+        {
+            // This test assumes broker v0.10.0 or higher:
+            // https://github.com/edenhill/librdkafka/wiki/Broker-version-compatibility
+
+            // This test does a broker metadata request, as it's an easy way to see 
+            // if we are connected to broker. It's not really the best way to test this
+            // (ideally, we would get from a working list of brokers to all brokers 
+            // that have changed IP) but this will be good enough.
+
+            var producerConfig = new Dictionary<string, object>
+            {
+                { "bootstrap.servers", "unknown" },
+                { "api.version.request", true }
+            };
+
+            var consumerConfig = new Dictionary<string, object>
+            {
+                { "group.id", Guid.NewGuid().ToString() },
+                { "bootstrap.servers", "unknown" },
+                { "session.timeout.ms", 6000 },
+                { "api.version.request", true }
+            };
+
+            using (var typedProducer = new Producer<Null, string>(producerConfig, null, new StringSerializer(Encoding.UTF8)))
+            {
+                TestMetadata(
+                    () => typedProducer.GetMetadata(false, null, TimeSpan.FromSeconds(3)),
+                    typedProducer.AddBrokers);
+            }
+
+            using (var producer = new Producer(producerConfig))
+            {
+                TestMetadata(
+                    () => producer.GetMetadata(false, null, TimeSpan.FromSeconds(3)),
+                    producer.AddBrokers);
+            }
+
+            using (var consumer = new Consumer(consumerConfig))
+            {
+                TestMetadata(
+                    () => consumer.GetMetadata(false, TimeSpan.FromSeconds(3)),
+                    consumer.AddBrokers);
+            }
+
+            using (var typedConsumer = new Consumer<Null, Null>(consumerConfig, new NullDeserializer(), new NullDeserializer()))
+            {
+                TestMetadata(
+                    () => typedConsumer.GetMetadata(false, TimeSpan.FromSeconds(3)),
+                    typedConsumer.AddBrokers);
+            }
+
+            void TestMetadata(Func<Metadata> getMetadata, Func<string, int> addBrokers)
+            {
+                try
+                {
+                    var metadata = getMetadata();
+                    Assert.True(false, "Broker should not be reached here");
+                }
+                catch (KafkaException e)
+                {
+                    Assert.Equal(ErrorCode.Local_Transport, e.Error.Code);
+                }
+
+                int brokersAdded = addBrokers(bootstrapServers);
+                Assert.True(brokersAdded > 0, "Should have added one broker or more");
```

Comment:
We're adding the same broker again here so I would have expected this to return 0. I'm guessing the return value is equal to the number of valid brokers in the supplied list regardless of whether they are already known or not. correct @edenhill ?